### PR TITLE
feat(eslint-plugin): deprecate no-loss-of-precision extension rule

### LINKED ISSUE #7277: 🔪 Deprecate no-loss-of-precision rule extension
## Overview

https://typescript-eslint.io/rules/no-loss-of-precision states that it exists to add support for [numeric separators](https://github.com/tc39/proposal-numeric-separator) to the base [eslint/no-loss-of-precision](https://eslint.org/docs/rules/no-loss-of-precision) rule. But, numeric separators have passed stage 3 and are natively supported in ESLint I believe. [[Example ESLint playground with the rule working on a number with a numeric separator](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tbG9zcy1vZi1wcmVjaXNpb246IFwiZXJyb3JcIiovXG5cbmNvbnN0IHggPSA5MDA3MTk5XzI1NDc0XzA5OTMiLCJvcHRpb25zIjp7ImVudiI6e30sInJ1bGVzIjp7fSwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQifX19)]

tl;dr: I propose we remove the rule _extension_ because the base rule contains all its features. 

👉 I'm not proposing we remove the base rule (or ask ESLint to)

> Whether it exists in popular configs such as eslint-config-airbnb-typescript and eslint-config-standard-with-typescript

* `eslint-config-airbnb-typescript`: yes, https://github.com/iamturns/eslint-config-airbnb-typescript/blob/8ef77c928c97d977f053c9c638831363a715d4a9/lib/shared.js#L143
* `eslint-config-standard-with-typescript`: yes, https://github.com/standard/eslint-config-standard-with-typescript/blob/b7b298d13ec737684e4368c726a3a9a66cd2f465/src/index.ts#L16

> Sourcegraph queries showing how often it appears in user configs

[Sourcegraph query showing it used in <200 repos](https://sourcegraph.com/search?q=context:global+file:.*eslint%7Cjs%7Cjson%7Cyaml%7Cyml%7Cyaml.*+%22%40typescript-eslint/no-loss-of-precision%22+-file:.*node_modules.*+-file:dist%5C/.*+count:200000&patternType=standard&sm=0&groupBy=repo): this doesn't seem like a particularly common rule for users to explicitly specify. That makes sense to me given that it's in the base recommended config (and has been for a while).

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
BREAKING CHANGE: Removes the rule from recommended-type-checked.

## PR Checklist

- [x] Addresses an existing open issue: fixes #7277
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Deprecates the rule and removes it from `recommended`.

💖

### CODE REVIEW

File: packages/eslint-plugin/src/rules/no-loss-of-precision.ts
Comment (kirkwaiblinger) [inline packages/eslint-plugin/src/rules/no-loss-of-precision.ts:1]: (optional) If you feel like really proving that this rule is redundant to the base rule, you could just literally proxy the base rule (passes all test cases for me). 

```diff
  create(context) {
-    const rules = baseRule.create(context);
-    function isSeparatedNumeric(node: TSESTree.Literal): boolean {
-      return typeof node.value === 'number' && node.raw.includes('_');
-    }
-    return {
-      Literal(node: TSESTree.Literal): void {
-        rules.Literal({
-          ...node,
-          raw: isSeparatedNumeric(node) ? node.raw.replace(/_/g, '') : node.raw,
-        } as never);
-      },
+    return baseRule.create(context);
    };
  },
```

**Review by kirkwaiblinger — APPROVED** [review]
Cast it into the fire!

File: packages/eslint-plugin/src/rules/no-loss-of-precision.ts
Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/src/rules/no-loss-of-precision.ts:1]: I like it!
